### ISSUE TITLE
a11y: contrast problem on hover - select-menu-list > select-menu-item

### ISSUE THREAD

**Issue by schalkneethling** [primer/react#3272]
### Description

I hope this is the correct repository for this report (I chose it based on a similar conversation here https://github.com/primer/react/issues/2229). I ran into a color contrast problem, or more specifically, a problem with the text color of the `unavailable-merge-method` class when a `select-menu-item` is hovered over or focused.

![Screenshot of the update branch UI button with its select menu expanded. It shows illegible text when a select menu item is hovered over or focused](https://github.com/primer/react/assets/10350960/111b71c0-117d-4bbd-916e-7cac9e0f9b8e)

The same problem can be seen when using the merge button `select-menu-list`:

![Screenshot of the merge pull request UI button with its select menu expanded. It shows illegible text when a select menu item is hovered over or focused](https://github.com/primer/react/assets/10350960/ed14ccdf-0f4f-42b4-80f2-33bb54b01731)

Interestingly, while the `unavailable-merge-method` class does not have a higher specificity than `.select-menu-item:focus` or `.select-menu-item:hover` it still overrides the color set for `.select-menu-item:focus`. This suggests that the `layout.scss` file is loaded after `syntax-theme-codemirror.scss`.

I assume something like this could help the file order problem, as I know it is not always possible to change the file import order.

```scss
.select-menu-item {
  &:hover,
  &:focus {
    .unavailable-merge-method {
      color: var(--color-foreground-hover);
    }
  }
}
``` 

### Sidenote

If you tab through the available options in a `select-menu-list`, the unavailable option is skipped over. I am not sure whether this is the best approach. The unavailable option does contain informative text ("Not enabled for this repository"), which is accessible to users without a visual impairment, even without the button being focusable.

> The reason the entry is skipped is because the button element that wraps the information is disabled the default browser behaviour is to skip over the button.

I tried to access the information using VoiceOver and I am not able to do so. I am not a daily screen reader user so, there might very well be an easy way to get to the information inside the unavailable option.

https://github.com/primer/react/assets/10350960/5591e912-dc97-4e30-9e5f-fb21f1d353f9

> The above video shows the behavior of the element when accessed using a keyboard and the feedback provided by the VoiceOver application on macOS.

### Steps to reproduce

1. Click on the down arrow to the right of either the update branch or merge button.
2. Hover over or focus (* more on the focus state in a minute) a select menu item that is not an available option for the current action.
3. The color of the text inside the `span` element with the class `unavailable-merge-method` does not change and is therefore almost illegible (see screenshots in the above description)

### Version

N/A

### Browser

Chrome, Safari, Firefox, Edge

**Comment by schalkneethling** [comment]
> The reason the entry is skipped is because the button element that wraps the information is disabled the default browser behaviour is to skip over the button.

I was thinking about this some more, and perhaps the solution to the unavailable option in the select list is for it _not_ to be a button. It could instead be a `div` with `tabindex=-1` so that it is focusable. I do not believe the user needs to know all the content in there, so one could use an `aria-label` that simply states something like "Rebase and merge is not enabled for this repository."

![Screenshot demonstrating what the hover/focus state can look like and how an aria label attribute can be used to quickly give an overview to a screen reader user.](https://github.com/primer/react/assets/10350960/90b70185-7d4a-41af-9b3d-89cd32c0d48e)

Something like the above. This _does_ mean the component needs to do some extra work to move focus to the `div` containers, which will not receive focus when using the tab key by default. With that said, it would be worth it I believe.

**Comment by lesliecdubs** [comment]
Thanks for the detailed report! This part of the GitHub UI isn't actually using a Primer React component, however this is a known bug that we have an internal issue filed for fixing. I'm going to close this for now but please let us know if you have any concerns.
